### PR TITLE
#7874 Add support for group attributes

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
This PR provides support for group attribute management. By default the tab is hidden (we can decide to add one notes attribute, just to test the functionality). 

In draft because it requires : 
 - [x] Unit tests
 - [x] Support on back-end. @taba90 is going to provide it. Issue on geostore TBD.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

No group attribute support

**What is the new behavior?**

#7874 

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/plugins/manager/GroupManager.jsx [hunk 1]
```diff
@@ -17,9 +17,10 @@ import Button from '../../components/misc/Button';
 import { editGroup, getUserGroups, groupSearchTextChanged } from '../../actions/usergroups';
 import Message from '../../components/I18N/Message';
 import SearchBar from '../../components/search/SearchBar';
-import GroupDeleteConfirm from './users/GroupDeleteConfirm';
-import GroupDialog from './users/GroupDialog';
-import GroupsGrid from './users/GroupGrid';
+import GroupDeleteConfirm from '../../plugins/manager/users/GroupDeleteConfirm';
+import GroupDialog from '../../plugins/manager/users/GroupDialog';
+import GroupsGrid from '../../plugins/manager/users/GroupGrid';
```
Comment (offtherailz) [inline web/client/plugins/manager/GroupManager.jsx:22]:
```suggestion
import GroupDeleteConfirm from './users/GroupDeleteConfirm';
import GroupDialog from './users/GroupDialog';
import GroupsGrid from './users/GroupGrid';
```

**Comment by tdipisa** [comment]
@offtherailz @taba90 moving to next major. We will include this on MS master as soon as also GeoStore master will be updated accordingly.

File: web/client/plugins/manager/GroupManager.jsx [hunk 2]
```diff
@@ -88,8 +95,93 @@ class GroupManager extends React.Component {
  * Allows an administrator to browse user groups.
  * Renders in {@link #plugins.Manager|Manager} plugin.
  * @name GroupManager
+ * @property {object[]} attributeFields attributes that should be shown in attributes tab of group manager.
```
Comment (offtherailz) [inline web/client/plugins/manager/GroupManager.jsx:98]:
```suggestion
 * @property {object[]} [attributeFields] attributes that should be shown in attributes tab of group manager.
```

File: docs/developer-guide/mapstore-migration-guide.md [hunk 3]
```diff
@@ -20,6 +20,17 @@ This is a list of things to check if you want to update from a previous version
...
+## Migration from 2022.02.xx to 2023.01.00
+
+### Update database schema
+
+This new version introduced the attributes for user groups. This requires an update to your database applying the scripts available [here](https://github.com/geosolutions-it/geostore/tree/master/doc/sql/migration). You have to apply the script `*-migration-from-v.1.5.0-to-v2.0.0` of your database. For instance on postgreSQL, you will have to execute the script `postgresql/postgresql-migration-from-v.1.5.0-to-v2.0.0`.
+
+!!! note:
+    The script assumes you set the search path for your db schema. Usually in postgres it is `geostore`. So make you sure to set the proper search path before to execute the script in postgres. (e.g. `SET search_path TO geostore;` )
+
+
```
Comment (offtherailz) [inline docs/developer-guide/mapstore-migration-guide.md:32]:
```suggestion
!!! node
    If you want or you can not execute the migration script, you can set in `geostore-datasource-ovr.properities` the following property to make MapStore update the database for you
    
    ```properties
    geostoreEntityManagerFactory.jpaPropertyMap[hibernate.hbm2ddl.auto]=update
    ```

```

File: docs/developer-guide/mapstore-migration-guide.md [hunk 4]
```diff
@@ -20,6 +20,23 @@ This is a list of things to check if you want to update from a previous version
...
+### Update database schema
+
+This new version introduced the attributes for user groups. This requires an update to your database applying the scripts available [here](https://github.com/geosolutions-it/geostore/tree/master/doc/sql/migration). You have to apply the script `*-migration-from-v.1.5.0-to-v2.0.0` of your database. For instance on postgreSQL, you will have to execute the script `postgresql/postgresql-migration-from-v.1.5.0-to-v2.0.0`.
+
+!!! note:
+    The script assumes you set the search path for your db schema. Usually in postgres it is `geostore`. So make you sure to set the proper search path before to execute the script in postgres. (e.g. `SET search_path TO geostore;` )
+
+!!! node
+    If you want or you can not execute the migration script, you can set in `geostore-datasource-ovr.properities` the following property to make MapStore update the database for you
```
Comment (offtherailz) [inline docs/developer-guide/mapstore-migration-guide.md:33]:
```suggestion
    If you don't want to or you can not execute the migration script, you can set in `geostore-datasource-ovr.properities` the following property to make MapStore update the database for you
```

File: docs/developer-guide/mapstore-migration-guide.md [hunk 5]
```diff
@@ -20,6 +20,23 @@ This is a list of things to check if you want to update from a previous version
...
+### Update database schema
+
+This new version introduced the attributes for user groups. This requires an update to your database applying the scripts available [here](https://github.com/geosolutions-it/geostore/tree/master/doc/sql/migration). You have to apply the script `*-migration-from-v.1.5.0-to-v2.0.0` of your database. For instance on postgreSQL, you will have to execute the script `postgresql/postgresql-migration-from-v.1.5.0-to-v2.0.0`.
+
+!!! note:
+    The script assumes you set the search path for your db schema. Usually in postgres it is `geostore`. So make you sure to set the proper search path before to execute the script in postgres. (e.g. `SET search_path TO geostore;` )
+
+!!! node
+    If you don't want to or you can not execute the migration script, you can set in `geostore-datasource-ovr.properities` the following property to make MapStore update the database for you
+
```
Comment (offtherailz) [inline docs/developer-guide/mapstore-migration-guide.md:34]:
```suggestion

```

File: docs/developer-guide/mapstore-migration-guide.md [hunk 6]
```diff
@@ -20,6 +20,23 @@ This is a list of things to check if you want to update from a previous version
...
+    The script assumes you set the search path for your db schema. Usually in postgres it is `geostore`. So make you sure to set the proper search path before to execute the script in postgres. (e.g. `SET search_path TO geostore;` )
+
+!!! node
+    If you don't want to or you can not execute the migration script, you can set in `geostore-datasource-ovr.properities` the following property to make MapStore update the database for you
+    
+    ```properties
+    geostoreEntityManagerFactory.jpaPropertyMap[hibernate.hbm2ddl.auto]=update
+    ```
+
+
```
Comment (offtherailz) [inline docs/developer-guide/mapstore-migration-guide.md:39]:
```suggestion
```

File: web/client/components/manager/users/UserDialog.jsx [hunk 7]
```diff
@@ -249,7 +263,7 @@ class UserDialog extends React.Component {
     };
 
     render() {
-        return (<Dialog modal draggable={false} maskLoading={this.props.user && (this.props.user.status === "loading" || this.props.user.status === "saving")} id="mapstore-user-dialog" className="user-edit-dialog" style={assign({}, this.props.style, {display: this.props.show ? "block" : "none"})}>
+        return (!this.props.show ? null : <Dialog modal draggable={false} maskLoading={this.props.user && (this.props.user.status === "loading" || this.props.user.status === "saving")} id="mapstore-user-dialog" className="user-edit-dialog" style={assign({}, this.props.style, {display: this.props.show ? "block" : "none"})}>
```
Comment (MV88) [inline web/client/components/manager/users/UserDialog.jsx:266]:
```suggestion
        return (!this.props.show ? null : <Dialog modal draggable={false} maskLoading={this.props.user && (this.props.user.status === "loading" || this.props.user.status === "saving")} id="mapstore-user-dialog" className="user-edit-dialog" style={this.props.style}>
```

File: web/client/plugins/manager/GroupManager.jsx [hunk 8]
```diff
@@ -88,8 +95,93 @@ class GroupManager extends React.Component {
...
  * @memberof plugins
  * @class
+ * @example
+ * { "name": "GroupManager",
+ *   "cfg": {
+ *        "showMembersTab": false,
+ *        "showAttributesTab": true,
+ *        "attributeFields": [
+ *             {
+ *                 "title": "Simple text",
```
Comment (MV88) [inline web/client/plugins/manager/GroupManager.jsx:110]: **NOTE:** 
for the future we may want to localize these labels, by using paths to translations ids and using Message comp in the UserDialog comp or GroupDialog comp

File: web/client/plugins/manager/GroupManager.jsx [hunk 9]
```diff
@@ -88,8 +95,93 @@ class GroupManager extends React.Component {
  * Allows an administrator to browse user groups.
  * Renders in {@link #plugins.Manager|Manager} plugin.
  * @name GroupManager
+ * @property {object[]} [attributeFields] attributes that should be shown in attributes tab of group manager.
```
Comment (MV88) [inline web/client/plugins/manager/GroupManager.jsx:98]: please specify defaults for attribute fields, like separator ecc

**Review by MV88 — CHANGES_REQUESTED** [review]
a few changes here and there

the rest is good

File: web/client/components/manager/users/AttributeControls.jsx [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+    const {url, path, valueAttribute = "value", labelAttribute = "label"} = source;
+    const [options, setOptions] = useState();
+    const loadOptions = () => axios.get(url).then(({data}) => {
+        return (path ? get(data, path) : data).map((entry = {}) => ({
+            value: entry?.[valueAttribute],
+            label: entry?.[labelAttribute]
+        }));
+    });
+    useEffect(() => {
+        loadOptions().then(setOptions);
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply web/client/components/manager/users/AttributeControls.jsx:61]: no options are show.  I think it is acceptable (I don't think react-select provides sufficient api for error show).

File: web/client/plugins/manager/GroupManager.jsx (hunk 9 above)
Comment (offtherailz) [reply web/client/plugins/manager/GroupManager.jsx:98]:
```suggestion
 * @property {object[]} [attributeFields] attributes that should be shown in attributes tab of group manager. By default this array contains one `notes` attribute with `controlType`: `text`. Every object in this array can contain: 
 * - `name`: the name of the attribute
 * - `title`: the string to show as label for the attribute.  If not present 
 * - `controlType`: The input control to use. can be : `string`, `text` (text area), `date` (date picker) and `select`. 
 * - `controlAttributes`: attributes specific of the `controlType`. See the examples for more details.
```

File: web/client/plugins/manager/GroupManager.jsx (hunk 8 above)
Comment (offtherailz) [reply web/client/plugins/manager/GroupManager.jsx:110]: yes, Good suggestion. If it will be required in the future, we can do a separate PR.

**Review by MV88 — APPROVED** [review]

**Comment by offtherailz** [comment]
@ElenaGallo, could you please test this on  DEV ? Thank you

as admin user you will be able to see attributes tab, with a notes attribute. You can test by creating a new group saving its attributes and so on.

**Pull request merged**